### DISCUSSION TITLE
the preprocessor for detection, only resizing

### DISCUSSION THREAD

**Discussion by mohamedsaeed8223** [mindee/doctr#1799] [Q&A]
I have some cases where the images' dimensions are far from being like a square,  their height are far larger than their width or the other way around

wouldn't it be better for those cases if the Document object or the preprocessor had the option to have them padded the image to the size we set to the model to take in?

maybe check the original width to original height ratio and if its too high or too low opt to padding the image

**Comment by felixdittrich92** [comment]
Hey @mohamedsaeed8223 👋 ,

That's already the case and enabled by default.
From docs:

```
preserve_aspect_ratio: if you want to preserve the aspect ratio of your documents while resizing before sending them to the model.

symmetric_pad: if you choose to preserve the aspect ratio, it will pad the image symmetrically and not from the bottom-right.
```

Both are set to `True` as default values.

Best,
Felix :)

**Reply by mohamedsaeed8223** [reply]
Thanks for replying @felixdittrich92 !💌

I found our key words `preserve_aspect_ratio` , `symmetric_pad` in the detection predictor class [here](https://github.com/mindee/doctr/blob/72bed4152a0228ed5b7ec56e990c5475937cd6a8/doctr/models/detection/predictor/pytorch.py#L44)

but they seem to be attributes of the `torchvision.transforms.Resize` object, and got no traces of them being set to `True`

although they could be passed as the extra `**kwargs` in the preprocessor constructor [here](https://github.com/mindee/doctr/blob/72bed4152a0228ed5b7ec56e990c5475937cd6a8/doctr/models/preprocessor/pytorch.py#L42)

but looking in the [torchvision docs](https://pytorch.org/vision/stable/_modules/torchvision/transforms/transforms.html#Resize) if the Resize object could have those , and It seems like it cant. and those are completely different augments with their separate classes.

It would be highly appreciated if you could refer me to the code that can set those parameters to True, or how I could set them to be true while still using the preprocessor and predictor APIs.

Edit: it looks like doctr's custom Resize is used instead of torchvision's , but the values of `preserve_aspect_ratio` and `symmetric_pad` values are set to false by default [here](https://github.com/mindee/doctr/blob/72bed4152a0228ed5b7ec56e990c5475937cd6a8/doctr/transforms/modules/pytorch.py#L27)

its clear now how I could set them to be true:)

Thanks

**Reply by felixdittrich92** [reply]
https://github.com/mindee/doctr/blob/72bed4152a0228ed5b7ec56e990c5475937cd6a8/doctr/transforms/modules/pytorch.py#L20 

About the args for the `detection_predictor` that's already updated to make it clearer for users on main branch:
https://github.com/mindee/doctr/blob/72bed4152a0228ed5b7ec56e990c5475937cd6a8/doctr/models/detection/zoo.py#L78

Hope this helps :)
